backend/src/middleware/hashtags/extractHashtags.js

```diff
@@ -4,23 +4,23 @@ import { exec, build } from 'xregexp/xregexp-all.js'
 //   https://en.wikipedia.org/w/index.php?title=Hashtag&oldid=905141980#Style
 // here:
 //    0. Search for whole string.
-//    1. Hashtag has only all unicode characters and '0-9'.
-//    2. If it starts with a digit '0-9' than a unicode character has to follow.
-const regX = build('^/search/hashtag/((\\pL+[\\pL0-9]*)|([0-9]+\\pL+[\\pL0-9]*))$')
+//    1. Hashtag has only all unicode letters and '0-9'.
+//    2. If it starts with a digit '0-9' than a unicode letter has to follow.
+const regX = build('^((\\pL+[\\pL0-9]*)|([0-9]+\\pL+[\\pL0-9]*))$')
 
 export default function(content) {
   if (!content) return []
   const $ = cheerio.load(content)
   // We can not search for class '.hashtag', because the classes are removed at the 'xss' middleware.
   //   But we have to know, which Hashtags are removed from the content as well, so we search for the 'a' html-tag.
-  const urls = $('a')
+  const ids = $('a[data-hashtag-id]')
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-17T06:31:01Z" title="Tuesday, September 17th 2019, 8:31:01 am +02:00">Sep 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -6,21 +6,21 @@ import { exec, build } from 'xregexp/xregexp-all.js'
 //    0. Search for whole string.
 //    1. Hashtag has only all unicode characters and '0-9'.
 //    2. If it starts with a digit '0-9' than a unicode character has to follow.
-const regX = build('^/search/hashtag/((\\pL+[\\pL0-9]*)|([0-9]+\\pL+[\\pL0-9]*))$')
+const regX = build('^((\\pL+[\\pL0-9]*)|([0-9]+\\pL+[\\pL0-9]*))$')
 
 export default function(content) {
   if (!content) return []
   const $ = cheerio.load(content)
   // We can not search for class '.hashtag', because the classes are removed at the 'xss' middleware.
   //   But we have to know, which Hashtags are removed from the content as well, so we search for the 'a' html-tag.
-  const urls = $('a')
+  const ids = $('a[data-hashtag-id]')
```


I like this change!

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-17T08:12:44Z" title="Tuesday, September 17th 2019, 10:12:44 am +02:00">Sep 17, 2019</time>_

---


@mattwr18 This change breaks the functionality. 😂